src/python/deepgnn/graph_engine/snark/tests/e2e_test.py

```diff
@@ -231,39 +247,120 @@ def __call__(self, x):
         return self.count % 2
 
 
+def linearize(value):
+    def get_features(item):
+        output = []
+        for feature_key, feature_dict in item.items():
+            if "feature" not in feature_key:
+                continue
+            for key, v in feature_dict.items():
+                while int(key) + 1 > len(output):
+                    output.append(None)
+                if feature_key == "binary_feature":
+                    vv = v
+                elif "sparse" in feature_key:
+                    vv = (
+                        np.array(v["coordinates"], dtype=np.int64),
+                        np.array(
+                            v["values"],
+                            dtype=JsonDecoder.convert_map[
+                                feature_key.replace("sparse_", "")
+                            ],
+                        ),
+                    )
+                else:
+                    vv = np.array(v, dtype=JsonDecoder.convert_map[feature_key])
+                output[int(key)] = vv
+        return output
+
+    if "node_id" in value:
+        return (
+            value["node_id"],
+            -1,
+            value["node_type"],
+            value["node_weight"],
+            get_features(value),
+        )
+    else:
+        return (
+            value["src_id"],
+            value["dst_id"],
+            value["edge_type"],
+            value["weight"],
+            get_features(value),
+        )
+
+
+def write_multi_binary(output_dir, partitions):
+    partition_meta = ""
+    for i, p in enumerate(partitions):
+        writer = BinaryWriter(output_dir, i)
+        writer.add([linearize(v) for v in p])
```

Comment:
That's not how `BinaryWriter` is supposed to work. From documentation:
```
data: Iterable[(int, int, int, float, list)] Data for a node first, then all of
                its edges in order of dst. Each entry for a node/edge is
```

There is nothing about splitting features and expected outcomes.

Reply:
Updated this usage.